### PR TITLE
Change hdiutil detach

### PR DESCRIPTION
**Pull request by eFrane**
The detach now always fetches the correct mountpoint.

### CODE REVIEW

**Comment by jpakkane** [comment]
Looks good, but please don't remove the comment. AFAICT the output format of `hdiutil` is not stable. Apple has changed it in the past and may do so again.

**Comment by eFrane** [comment]
I readded that comment. I also tried using `hdiutil`'s `-puppetstrings`-option which supposedly will always return coherent output, unfortunately that does not seem to work with `hdiutil info`. Another save approach might be to use `diskutil` for device discovery since the device names are the same. The important thing afaict is to detach with `hdiutil` to make sure the disk image is closed in a clean way.

File: manual tests/4 standalone binaries/build_osx_package.sh [hunk 1]
```diff
@@ -8,13 +8,12 @@ rm -rf buildtmp
 mkdir -p mnttmp
 rm -f working.dmg
 gunzip < template.dmg.gz > working.dmg
-hdiutil attach working.dmg -noautoopen -quiet -mountpoint mnttmp
 # NOTE: output of hdiutil changes every now and then.
 # Verify that this is still working.
-DEV=`hdiutil info|tail -1|awk '{print $1}'`
+hdiutil attach working.dmg -noautoopen -quiet -mountpoint mnttmp
```
Comment (QuLogic) [inline manual tests/4 standalone binaries/build_osx_package.sh]: This got moved after the comment, but hasn't actually changed and might not be related to the comment.

File: manual tests/4 standalone binaries/build_osx_package.sh [hunk 2]
```diff
@@ -8,13 +8,12 @@ rm -rf buildtmp
...
 rm -rf mnttmp/myapp.app
 mv /tmp/myapp.app mnttmp
-hdiutil detach ${DEV}
+hdiutil detach $(hdiutil info|grep "mnttmp"|awk '{print $1}')
```
Comment (QuLogic) [inline manual tests/4 standalone binaries/build_osx_package.sh]: The comment refers to the stuff in `$()`, but it's now quite a bit away from what it's referencing.

**Comment by jpakkane** [comment]
I merged this manually with the fixes. Thanks.

**Pull request closed without merging**